client/src/app/pages/assessment/import-questionnaire-form/import-questionnaire-form.tsx

```diff
@@ -0,0 +1,177 @@
+
+export interface ImportQuestionnaireFormProps {
+  onSaved: (response: AxiosResponse) => void;
+}
+export interface ImportQuestionnaireFormValues {
+  yamlFile: IReadFile;
+}
+
+export const yamlFileSchema: yup.SchemaOf<IReadFile> = yup.object({
+  fileName: yup.string().required(),
+  fullFile: yup.mixed<File>(),
+  loadError: yup.mixed<FileLoadError>(),
+  loadPercentage: yup.number(),
+  loadResult: yup.mixed<"danger" | "success" | undefined>(),
+  data: yup.string(),
+  responseID: yup.number(),
+});
+
+export const ImportQuestionnaireForm: React.FC<
+  ImportQuestionnaireFormProps
+> = ({ onSaved }) => {
+  const { t } = useTranslation();
+  const { pushNotification } = React.useContext(NotificationsContext);
+
+  const [filename, setFilename] = React.useState<string>();
+  const [isFileRejected, setIsFileRejected] = useState(false);
+  const validationSchema: yup.SchemaOf<ImportQuestionnaireFormValues> = yup
+    .object()
+    .shape({
+      yamlFile: yamlFileSchema,
+    });
+  const methods = useForm<ImportQuestionnaireFormValues>({
+    resolver: yupResolver(validationSchema),
+    mode: "onChange",
+  });
+
+  const {
+    handleSubmit,
+    formState: { isSubmitting, isValidating, isValid, isDirty },
+    getValues,
+    setValue,
+    control,
+    watch,
+    setFocus,
+    clearErrors,
+    trigger,
+    reset,
+  } = methods;
+
+  const { mutateAsync: createYamlFileAsync } = useCreateFileMutation();
+
+  const handleFileUpload = async (file: File) => {
+    setFilename(file.name);
+    const formFile = new FormData();
+    formFile.append("file", file);
+
+    const newYamlFile: IReadFile = {
+      fileName: file.name,
+      fullFile: file,
+    };
+
+    return createYamlFileAsync({
+      formData: formFile,
+      file: newYamlFile,
+    });
+  };
+
+  const onSubmit = (values: ImportQuestionnaireFormValues) => {
+    console.log("values", values);
+  };
+  return (
+    <Form onSubmit={handleSubmit(onSubmit)}>
+      <HookFormPFGroupController
+        control={control}
+        name="yamlFile"
+        label={t("terms.uploadYamlFile")}
+        fieldId="yamlFile"
+        helperText="Upload a YAML file"
+        renderInput={({ field: { onChange, name }, fieldState: { error } }) => (
+          <FileUpload
+            id="yamlFile"
+            name={name}
+            value={filename}
+            filename={filename}
+            filenamePlaceholder="Drag and drop a file or upload one"
+            dropzoneProps={{
+              accept: {
+                "text/yaml": [".yml", ".yaml"],
+                // "text/xml": [".xml"],
+              },
+              maxSize: 1000000,
+              onDropRejected: (event) => {
+                const currentFile = event[0];
+                if (currentFile.file.size > 1000000) {
+                  methods.setError("yamlFile", {
+                    type: "custom",
+                    message: "Max file size of 1 MB exceeded.",
```

Comment:
Same here, Is this text shown to the user? If so I think it should be translated as well